internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -144,9 +135,15 @@ type MongoDBScope struct {
 	Type string `mapstructure:"name"`
 }
 
+type AwsAssumeRole struct {
```

Comment:
Any good reason to redefine the API structures when they're available via `preview_models.Secrets20231128AssumeRoleRequest` and such? I believe we could unmarshal/decode directly into them instead of redefining them unless I'm missing something.

Reply:
Hmm, I'm wondering if this would be a bit easier to chat on in a Zoom call. There's a follow-up ticket [here](https://hashicorp.atlassian.net/browse/VAULT-29636) that will move from yaml to hcl/json, so the unmarshaling will change a bit very soon when that switches over (picking up that ticket right after this one). As for redefining structs as opposed to using the proto messages directly, seems like the complexity comes in because the bytes read from a file do not fit the schema of the proto messages perfectly. For example, the `SecretConfig` that users are expected to provide has some top level fields that won't fit provider-specific request bodies because those details are nested in the `details` field of the config file, which I'm accessing after it's been parsed. Unless I'm missing something obvious 🤔.

Reply:
Chatted a bit on a sync, we're going to be changing to hcl/json in the [next ticket](https://hashicorp.atlassian.net/browse/VAULT-29636), so we don't want to add to much work here considering it would be undone in the next PR. Because the `SecretConfig` doesn't map to the proto messages, we'll have to see what's offered (if anything) via some hcl libraries to try and unmarshal/decode directly from the hcl file to request bodies.